apollo-federation/src/sources/connect/mod.rs

```diff
@@ -210,3 +450,558 @@ pub struct ConnectFetchNode {
+    use crate::schema::position::ScalarTypeDefinitionPosition;
+    use crate::schema::ValidFederationSchema;
+    use crate::source_aware::federated_query_graph::graph_path::OperationPathElement;
+    use crate::source_aware::federated_query_graph::FederatedQueryGraph;
+    use crate::source_aware::federated_query_graph::FederatedQueryGraphEdge;
+    use crate::source_aware::federated_query_graph::FederatedQueryGraphNode;
+    use crate::sources::connect::ConnectFederatedEnumQueryGraphNode;
+    use crate::sources::connect::ConnectFederatedScalarQueryGraphNode;
+    use crate::sources::connect::ConnectPathField;
+    use crate::sources::connect::ConnectPathSelections;
+    use crate::sources::connect::Selection;
+    use crate::sources::SourceFederatedConcreteFieldQueryGraphEdge;
+    use crate::sources::SourceFederatedConcreteQueryGraphNode;
+    use crate::sources::SourceFederatedEnumQueryGraphNode;
+    use crate::sources::SourceFederatedScalarQueryGraphNode;
+    use crate::sources::SourceId;
+    use crate::sources::SourcePathApi;
+
+    struct SetupInfo {
+        graph: DiGraph<FederatedQueryGraphNode, FederatedQueryGraphEdge>,
+        schema: ValidFederationSchema,
+        source_id: SourceId,
+    }
+    fn setup() -> SetupInfo {
+        let mut graph = DiGraph::new();
+        let source_id = SourceId::Connect(ConnectId {
+            label: "test connect".to_string(),
+            subgraph_name: "CONNECT".into(),
+            directive: ObjectOrInterfaceFieldDirectivePosition {
+                field: ObjectOrInterfaceFieldDefinitionPosition::Object(
+                    ObjectFieldDefinitionPosition {
+                        type_name: name!("TestObject"),
+                        field_name: name!("testField"),
+                    },
+                ),
+                directive_name: name!("connect"),
+                directive_index: 0,
+            },
+        });
+
+        // Create a dummy schema for tests
+        let schema = Schema::parse(include_str!("./tests/schemas/simple.graphql"), "").unwrap();
+        let schema = schema.validate().unwrap();
+        let schema = ValidFederationSchema::new(schema).unwrap();
+
+        // Fill in some dummy data
+        graph.add_node(FederatedQueryGraphNode::Concrete {
+            supergraph_type: ObjectTypeDefinitionPosition {
+                type_name: name!("Query"),
+            },
+            field_edges: IndexMap::new(),
+            source_exiting_edge: None,
+            source_id: source_id.clone(),
+            source_data: SourceFederatedConcreteQueryGraphNode::Connect(
+                ConnectFederatedConcreteQueryGraphNode::SelectionRoot {
+                    subgraph_type: ObjectTypeDefinitionPosition {
+                        type_name: name!("Query"),
+                    },
+                    property_path: Vec::new(),
+                },
+            ),
+        });
+
+        let source_id = SourceId::Connect(ConnectId {
```

Comment:
is there a reason this is defined twice?

Reply:
Whoops, good catch! Removed.